resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -68,6 +68,8 @@ private[spark] abstract class YarnSchedulerBackend(
   // Flag to specify whether this schedulerBackend should be reset.
   private var shouldResetOnAmRegister = false
 
+  private var lastRequestExecutors = RequestExecutors(-1, -1, Map.empty, Set.empty)
```

Comment:
I would suggest to change the first `-1` to `Utils#getDynamicAllocationInitialExecutors`, and the second to be "0".